front/lib/resources/storage/models/spaces.ts

```diff
@@ -66,3 +71,25 @@ SpaceModel.belongsTo(Workspace, {
   foreignKey: { allowNull: false },
   onDelete: "RESTRICT",
 });
+
+SpaceModel.addHook(
```

Comment:
Do we use partial index already ? It fit the purpose at the DB level (which is always better imho)

`CREATE UNIQUE INDEX unique_workspace_type 
ON your_table (workspaceId, kind)
WHERE type IN ('global', 'system', 'conversation');`

Reply:
I see the point. Counter argument:
- indexes add overhead, their purpose is to index, not enforce a constraint;
- here, we don't really need the index, we need to enforce a constraint, so using an index for that purpose feels hackish
- consistency with groups (we could switch groups too though, just imo out of scope)
There would also be the option of adding a regular SQL CONSTRAINT rather than an index, that might work?
In any case, i'll leave that for a follow up PR

Reply:
> indexes add overhead, their purpose is to index, not enforce a constraint;

I kinda disagree on that one, we do use a lot of classic unique indexes for both the index and the constraint.

One can argue that looking up the conversation, global or system spaces (a very common need) will be able to rely on this very fast index but I see your point.

> There would also be the option of adding a regular SQL CONSTRAINT rather than an index, that might work?

It's much more involved.

I don't mind not doing it, it was a suggestion. I don't like enforcing datastructure in the application layer (only).

Reply:
On that one, I understand your point but I'd rather leave it as is for now